src/commands/chat/ui/ChatUi.tsx

```diff
@@ -62,18 +53,62 @@ export function ChatUi() {
     }
   };
 
+  const showInput = !isLoading && !shouldExit;
+
   return (
     <Box display="flex" flexDirection="column">
       <ChatMessageList />
 
       {activeView === 'help' && <HelpOutput />}
       {activeView === 'info' && <InfoOutput />}
 
-      {loadingResponse ? <AiResponseLoader /> : null}
-      {!loadingResponse && !shouldExit && <UserMessageInput onSubmit={handleSubmit} />}
+      {isLoading ? <AiResponseLoader text={loadedResponse} /> : null}
+      {showInput && <UserMessageInput onSubmit={handleSubmit} />}
 
       <StatusBar />
       {shouldExit && <ExitApp />}
     </Box>
   );
 }
+
+function useAiResponse() {
```

Comment:
Can we extract this function to separate "hooks" folder?

Reply:
tbh, it's quite tightly coupled with that screen & related Zustand state, and not really re-usable imo